fixes #18853: provide support of array attributes in multipart/form-data request for axios and fetch typescript clients

jeandonaldroselin opened the pull request:
<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Please review my PR :
@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @davidgamero @mkusaka

macjohnny commented:
@jeandonaldroselin can you merge the most recent master and fix the CI issues if they still occur?

jeandonaldroselin commented:
> @jeandonaldroselin can you merge the most recent master and fix the CI issues if they still occur?

Hello @macjohnny,
I will check the state ASAP and push the changes 👍🏽

jeandonaldroselin commented:
> @jeandonaldroselin can you merge the most recent master and fix the CI issues if they still occur?

Hello @macjohnny,

Sorry I was busy a lot since last week.
I updated my PR with the latest master and updated the tests.
The related build just passed :).

Please can you have a look at it ?

macjohnny commented on samples/client/echo_api/typescript-axios/build/api.ts:
```diff
@@ -615,7 +615,7 @@ export const BodyApiAxiosParamCreator = function (configuration?: Configuration)
             if (files) {
                 files.forEach((element) => {
-                  localVarFormParams.append('files', element as any);
+                    localVarFormParams.append('files', element as any);
```
this looks unrelated

jeandonaldroselin replied:
Hello,

Yes it looks unrelated.
I had the build failing and I had to run 
the command : 

bash bin/generate-samples.sh 

after this run, this command only added this space change.
